### PR TITLE
change checkout confirm hook, add order_product_id in checkout model and  change key to product key in order class

### PR DESCRIPTION
**Pull request by handoyo**

### CODE REVIEW

**Comment by abantecart** [comment]
Hello @handoyo 
Thank you for the commit. 
We are planning the release of v1.3.4 today/tomorrow and this commit will not make it. 
We will repoint this one to v1.3.5 soon.

File: public_html/core/lib/order.php [hunk 1]
```diff
@@ -341,7 +341,7 @@ public function buildOrderData($indata)
 
         foreach ($this->cart->getProducts() + $this->cart->getVirtualProducts() as $key => $product) {
             $product_data[] = [
-                'key'             => $key,
+                'key'             => $product['key'],
```
[the remark this answers is not in the document]
Comment (handoyo) [reply public_html/core/lib/order.php:344]: i see the $key was not the product key but it is the array key, not the product key that was generated by using product id and options ($product_id.':'.md5(serialize($options));)

**Comment by handoyo** [comment]
Hi core devs, should i create new pull request for v1.3.5?

**Comment by abolabo** [comment]
hi. Yes. Current dev branch is 1.3.5

**Pull request closed without merging**